parser/parser.c3

```diff
@@ -913,11 +914,51 @@ function parse_function_call : (cursor : Cursor&, prefix : Expression) -> Functi
 }
 
 function is_likely_the_start_of_a_specialized_function : (cursor : Cursor&) -> bool = {
-    // XXX: this is a stub
+    // This function does an approximate forward lookahead to determine whether
+    // the cursor is currently at the start of a less than expression or a
+    // specialized function call.
+
+    // Since it does not do a full parse, it may incorrectly identify a legal
+    // less than operator an illegal function call, the user can always correct
+    // this with parens.
+
     if !cursor:is_next("<") {
         return false;
     }
-    return true;
+
+    let brackets_stack : Vector<u8> = make_vector<u8>(&cursor.allocator);
+    let lookahead : Cursor = cursor;
+    while !lookahead:is_next("\0") {
+        if is_one_of(lookahead:get_char(), "<"[0], "("[0], "{"[0], "["[0]) {
```

Comment:
Curious to see what this function call optimizes to

Reply:
I'll check... But it's full of dynamic allocation so not great (I think it's opaque to llvm until we annotate it). I could defo make it faster by removing the allocation and making it recursive, but unless it's a bottleneck I don't think it's worth the risk of a stack overflow.

Reply:
I'm curious about `is_one_of` too, I hope it's equivalent to whatever c produces with a switch block.

Reply:
I've checked `is_one_of` with a bunch of `int`s... If they are all known at compile time, then LLVM sees right through it and computes the answer for you. The recursive function calls are all inlined into the top-level call to `is_one_of`, resulting in a series of `cmp/je` instructions.